tests/database/browser/crawler_support.test.ts

```diff
@@ -0,0 +1,186 @@
+import { expect } from "chai";
+import { forceRestClient } from "../../../src/database/api/test_access";
+
+import { 
+  getRandomNode,
+  testAuthTokenProvider,
+  getFreshRepoFromReference
+} from "../helpers/util";
+
+// Some sanity checks for the ReadonlyRestClient crawler support.
+describe('Crawler Support', function() {
+  var initialData;
+  var normalRef;
+  var restRef;
+  var tokenProvider;
+
+  beforeEach(function(done) {
+    normalRef = getRandomNode();
+
+    forceRestClient(true);
+    restRef = getFreshRepoFromReference(normalRef);
+    forceRestClient(false);
+
+    tokenProvider = testAuthTokenProvider(restRef.database.app);
+
+    setInitialData(done);
+  });
+
+  afterEach(function() {
+    tokenProvider.setToken(null);
+  });
+
+  function setInitialData(done) {
+    // Set some initial data.
+    initialData = {
+      leaf: 42,
+      securedLeaf: 'secret',
+      leafWithPriority: { '.value': 42, '.priority': 'pri' },
+      obj: { a: 1, b: 2 },
+      list: {
+        10: { name: 'amy', age: 75, '.priority': 22 },
+        20: { name: 'becky', age: 42, '.priority': 52 },
+        30: { name: 'fred', age: 35, '.priority': 23 },
+        40: { name: 'fred', age: 29, '.priority': 26 },
+        50: { name: 'sally', age: 21, '.priority': 96 },
+        60: { name: 'tom', age: 16, '.priority': 15 },
+        70: { name: 'victor', age: 4, '.priority': 47 }
+      },
+      valueList: {
+        10: 'c',
+        20: 'b',
+        30: 'e',
+        40: 'f',
+        50: 'a',
+        60: 'd',
+        70: 'e'
+      }
+    };
+
+    normalRef.set(initialData, function(error) {
+      expect(error).to.equal(null);
+      done();
+    });
+  }
+
+  it('set() is a no-op', function(done) {
+    // This test mostly exists to make sure restRef really is using ReadonlyRestClient
+    // and we're not accidentally testing a normal Firebase connection. It also can
+    // be a little slow so adding an extra timeout to help out.
+    this.timeout(3500);
```

Comment:
I would prefer that we set a global timeout for these tests rather than individual ones.